module/module.go

```diff
@@ -196,7 +195,11 @@ func NewModule(ctx context.Context, address string, logger logging.Logger) (*Mod
 		grpc.UnaryInterceptor(grpc_middleware.ChainUnaryServer(unaries...)),
 		grpc.StreamInterceptor(grpc_middleware.ChainStreamServer(streams...)),
 	}
+
+	cancelCtx, cancel := context.WithCancel(context.Background())
 	m := &Module{
+		ctx:                   cancelCtx,
```

Comment:
IIUC, this is the context that represents "are we in shutdown". What you did here is normal w.r.t everything else written at Viam, but in my newfound criticism of "contexts everywhere" can I entice you to in giving this a different name? I'm not guaranteeing it'll be an improvement long-term but I'd like to experiment.

So my suggestion would be:
```
isAliveCtx, shutdownFn := context.WithCancel
m := &Module{
  isAliveCtx: isAliveCtx,
  shutdownFn: shutdownFn,
...
```

Or maybe `isAliveCtx` -> `shutdownCtx`?

The goal is to better label that this context is more of a service/process top-level context. And that not _anyone_ will simply invoke `cancelFn()`, but rather it's only invoked when the module is told to shutdown. I want to believe that contexts represent lifetimes of _actors_ that have _names_. And using those names when available would improve readability.